main/src/ca/uwaterloo/flix/language/phase/ClosureConv.scala

```diff
@@ -266,8 +266,9 @@ object ClosureConv {
     case Expression.NewObject(clazz, tpe, purity, methods0, loc) =>
       val methods = methods0 map {
         case JvmMethod(ident, fparams, exp, retTpe, purity, loc) =>
-          val closure = mkLambdaClosure(fparams, exp, retTpe, loc)
-          JvmMethod(ident, fparams, closure, retTpe, purity, loc)
+          val cloType = Type.mkPureUncurriedArrow(fparams.map(_.tpe), retTpe, loc)
```

Comment:
I think it should be an Impure arrow type. (Otherwise the optimizer may wreck havoc).